Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -8,55 +10,63 @@ public GildedRose(Item[] items) {
     }
 
     public void updateQuality() {
-        for (int i = 0; i < items.length; i++) {
-            if (!items[i].name.equals("Aged Brie")
-                    && !items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
-                    }
-                }
+        for (int days = 0; days < items.length; days++) 
+        {
+            if(items[days].name.equals("Sulfuras, Hand of Ragnaros")) {
+                updateHandOfRagnaros(items[days]);
```

Comment:
positive: much better structure than before, still if a new item is added you always need a new else-if....